alica_engine/include/engine/AlicaClock.h

```diff
@@ -128,10 +129,15 @@ class AlicaTime
 class AlicaClock
 {
 public:
-    AlicaClock() {}
-    virtual ~AlicaClock() {}
+    AlicaClock(AlicaEngine* ae)
+    :ae(ae) {};
+    AlicaClock () {};
+    virtual ~AlicaClock() {};
     virtual AlicaTime now() const;
-    void sleep(const AlicaTime&) const;
+    virtual void sleep(const AlicaTime&) const;
+
+protected:
+    AlicaEngine* ae;
```

Comment:
why do we need AlicaEngine reference in this class?

Reply:
What I think that If We want to give option to user select then it will need engine parsing because without it there was error in build.

Reply:
Sorry didn't get that. What compilation error are you getting without this?

Reply:
My bad, it has no use. I will remove that.
And also in the test we have to change the clock portion, that's why it is failing CI.

Reply:
yup you can call setClock on alicaContext now rather than engine.

Reply:
Are you talking about the test or removing the engine from the AlicaClock?

Reply:
In "test_agent_dies.cpp" you can "call setClock on alicaContext now rather than engine."